File: miniJobs/Domain/Entities/Applicant.cs
﻿using Domain.Interfaces;

namespace Domain.Entities;
[Table("applicants")]
public class Applicant : IEntity<int>
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("cv")]
    public byte[]? Cv { get; set; }

    [Column("experience")]
    public string? Experience { get; set; }

    [Column("description")]
    public string? Description { get; set; }

    [Column("wage_proposal")]
    public decimal? WageProposal { get; set; }

    public User User { get; set; }
    public ICollection<ApplicantJobType> ApplicantJobTypes { get; set; }
    [NotMapped]
    public decimal AverageRating { get; set; }
    [NotMapped]
    public int NumberOfFinishedJobs { get; set; }
}

File: miniJobs/UI/minijobs_admin/lib/models/applicant/applicant.dart
import 'dart:convert';
import 'dart:typed_data';
import 'package:decimal/decimal.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:minijobs_admin/models/city.dart';
import 'package:minijobs_admin/models/job_type.dart';
import 'package:minijobs_admin/models/user/user.dart';

import '../../enumerations/job_application_status.dart';

part 'applicant.g.dart';
@JsonSerializable()
class Applicant extends User {
  Uint8List? cv;
  String? description;
  String? experience;
  Decimal? wageProposal;
  JobType? jobType;
  int? numberOfFinishedJobs;
  Decimal? averageRating;
  List<JobType>? jobTypes;
  int?  jobApplicationId;
  bool? isRated;
  JobApplicationStatus? applicationStatus;

  Applicant(
      super.id,
      super.firstName,
      super.lastName,
      super.fullName,
      super.email,
      super.phoneNumber,
      super.role,
      super.deleted,
      super.accountConfirmed,
      super.photo,
      super.cityId,
      super.city,
      this.cv,
      this.description,
      this.experience,
      this.wageProposal,
      this.jobType,
      this.numberOfFinishedJobs,
      this.averageRating,
      this.jobTypes,
      this.jobApplicationId,
      this.isRated,
      this.applicationStatus);

  factory Applicant.fromJson(Map<String, dynamic> json) => _$ApplicantFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$ApplicantToJson(this);
}